### PR TITLE
Prepare for 1.21.0 release

### PR DESCRIPTION
**Pull request by jack-berg**

### CODE REVIEW

File: bom/build.gradle.kts [hunk 1]
```diff
@@ -13,4 +13,4 @@ otelBom.addFallback("opentelemetry-exporter-jaeger-proto", "1.17.0")
 otelBom.addFallback("opentelemetry-extension-annotations", "1.18.0")
 otelBom.addFallback("opentelemetry-sdk-extension-resources", "1.19.0")
 otelBom.addFallback("opentelemetry-sdk-extension-aws", "1.19.0")
-otelBom.addFallback("opentelemetry-extension-aws", "1.20.0")
+otelBom.addFallback("opentelemetry-extension-aws", "1.20.1")
```
Comment (trask) [inline bom/build.gradle.kts:16]: is this right?

**Review by mateuszrzeszutek — APPROVED** [review]

File: bom/build.gradle.kts (hunk 1 above)
Comment (mateuszrzeszutek) [reply bom/build.gradle.kts:16]: It is the latest release in maven central

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,86 @@
...
+
+#### Testing
+
+* Make APIs for asserting attributes consistent
+  ([#4882](https://github.com/open-telemetry/opentelemetry-java/pull/4882))
+
+### Semantic Conventions
+
+* Update semantic and resource attributes for spec v0.15.0
+  ([#4938](https://github.com/open-telemetry/opentelemetry-java/pull/4938))
```
Comment (jkwatson) [inline CHANGELOG.md:70]: we're up to v1.16.0, right?

File: bom/build.gradle.kts (hunk 1 above)
Comment (trask) [reply bom/build.gradle.kts:16]: oh sorry! I misread as `1.21.0` 🧐

**Pull request merged**